### PR TITLE
Bug 1830045 - New search bookmarks UI tests

### PR DESCRIPTION
**Pull request by AndiAJ**
Created new search bookmarks related UI tests based on our [manual tests](https://testrail.stage.mozaws.net/index.php?/suites/view/3192&group_by=cases:section_id&group_id=274060&group_order=asc).
All tests successfully passed 100x on Firebase.

### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1830045

### CODE REVIEW

File: fenix/app/src/androidTest/java/org/mozilla/fenix/ui/BookmarksTest.kt [hunk 1]
```diff
@@ -741,4 +748,115 @@ class BookmarksTest {
...
+    fun verifyVoiceSearchInBookmarksTest() {
+        val defaultWebPage = TestAssetHelper.getGenericAsset(mockWebServer, 1)
+
+        browserScreen {
+            createBookmark(defaultWebPage.url)
+        }.openThreeDotMenu {
+        }.openBookmarks {
+            clickSearchButton()
+            verifyBookmarksSearchBar(true)
+            clickVoiceSearchButton()
```
Comment (sv-ohorvath) [inline fenix/app/src/androidTest/java/org/mozilla/fenix/ui/BookmarksTest.kt:825]: You could add here a transition to SearchRobot and reuse the startVoiceSearch method. Maybe with a little tweaking: 
`fun startVoiceSearch() {
        voiceSearchButton.click()
}`

then use TestHelper.grantPermission()
then TestHelper.assertNativeAppOpens()

you can also do this refactor in toggleVoiceSearchTest()

File: fenix/app/src/androidTest/java/org/mozilla/fenix/helpers/TestHelper.kt [hunk 2]
```diff
@@ -273,6 +274,26 @@ object TestHelper {
 
     fun assertYoutubeAppOpens() = intended(toPackage(YOUTUBE_APP))
 
+    fun assertVoiceSearchAppOpens() {
```
Comment (sv-ohorvath) [inline fenix/app/src/androidTest/java/org/mozilla/fenix/helpers/TestHelper.kt:277]: you can reuse the grantSystemPermission() method and assertNativeAppOpens() maybe

**Review by AaronMT — APPROVED** [review]
LGTM based on Oana's comments

**Comment by AndiAJ** [comment]
Hmmm, `youControlYourDataCardTest` UI test seems to be failing all of a sudden, will need to fix it

**Comment by AndiAJ** [comment]
Fixed it in a separate PR and rebased Main.

**Review by sv-ohorvath — APPROVED** [review]

**Pull request merged**